Pin openshift-golang-builder to 1.23 rhel8 tag

grdryn opened the pull request:
In 4f3ee0f (#177), 1.23 was introduced, but just referenced by manifest digest. Unfortunately the Konflux bot appears to have assumed that it was the `latest` tag, and saw a "new" (read: several years old) digest for that tag, and bumped to it in 3798127 (#178) .

This change specifies the rhel_8_golang_1.23 tag, so that it will only bump to newer versions of _that_ tag.

This change also changes the registry domain to be brew.registry.redhat.io, since 68487ea (#194) appears to indicate that that's preferred/required for Konflux. One can use the `--build-arg` CLI option to use the registry-proxy locally as needed.

grdryn committed:
Pin openshift-golang-builder to 1.23 rhel8 tag

In 4f3ee0f, 1.23 was introduced, but just referenced by manifest
digest. Unfortunately the Konflux bot appears to have assumed that it
was the `latest` tag, and saw a "new" (read: several years old) digest
for that tag, and bumped to it in 3798127.

This change specifies the rhel_8_golang_1.23 tag, so that it will only
bump to newer versions of _that_ tag.

This change also changes the registry domain to be
brew.registry.redhat.io, since 68487ea (#194) appears to indicate that
that's preferred/required for Konflux. One can use the `--build-arg`
CLI option to use the registry-proxy locally as needed.

grdryn committed:
Fix whitespace in renovate.json for pre-commit

grdryn commented:
To clarify, #178 bumped the golang builder image from a Go 1.23 image back in time to a Go 1.17 image from 2021 (the `latest` image in that image repo is not maintained). :robot: :fire:

chipspeak reviewed (approved):
LGTM and thanks, Ger. Pesky bots!

MohammadiIram commented on Dockerfile.konflux:
```diff
@@ -2,7 +2,7 @@
 ARG SOURCE_CODE=.
 
 # BEGIN -- workaround lack of go-toolset for golang 1.23
-ARG GOLANG_IMAGE=registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder@sha256:b848cf648a383b6e207ebd150afc7e65a90efe0d7734f4f6d2767144d58d68d0
```
This should go away we cannot use Alias in FROM as a base image instead it should be
`
FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.23@sha256:ca0c771ecd4f606986253f747e2773fe2960a6b5e8e7a52f6a4797b173ac7f56 AS golang`

similar to [this](https://github.com/red-hat-data-services/codeflare-operator/blob/rhoai-2.18/Dockerfile.konflux#L5)

grdryn replied:
@MohammadiIram hmm, interesting :thinking: Can you share a link to where it's documented that we can't use this standard Dockerfile feature? I'm interested to understand why, and if there are any other features that we should avoid.
An example of an error that this causes might also be useful, if it causes one.

grdryn replied:
I've updated this in a new commit here now, but I'd still like to understand it if you can provide any more information :slightly_smiling_face:

MohammadiIram replied:
@grdryn I'm not sure if this is documented in the Konflux docs, when using an alias, and the results are different but the build fails . I'll try to find a log and attach it here.

grdryn committed:
Don't use ARG in FROM line in Dockerfile.konflux

See here for more info:
https://github.com/red-hat-data-services/codeflare-operator/pull/227#discussion_r1955839203

grdryn commented:
@MohammadiIram @sutaakar @chipspeak anyone plan to merge this? I don't have that ability
